Bottom: c1face496bb74d975ba8362bd5491ccc016ff73e
Top:    5693cb13b1c08838af784ea99a36558bb0dbd7de
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-02-10 12:54:56 +0000

Refresh of nearly-working

---

diff --git a/src/polygon1.c b/src/polygon1.c
index 7557a66..fdc52f9 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -2637,6 +2637,12 @@ poly_Boolean_free (POLYAREA * ai, POLYAREA * bi, POLYAREA ** res, int action)
       M_POLYAREA_separate_isected (&e, res, &holes, &a_isected);
       M_POLYAREA_label_separated (a_isected, b, FALSE);
       M_POLYAREA_Collect_separated (&e, a_isected, res, &holes, action, FALSE);
+      /* delete of a_isected which is left */
+      while ((p = a_isected) != NULL)
+      {
+        a_isected = p->next;
+        poly_DelContour (&p);
+      }
       M_B_AREA_Collect (&e, b, res, &holes, action);
       poly_Free (&b);
